iroh-willow/src/net/codec.rs

```diff
@@ -0,0 +1,55 @@
+use anyhow::ensure;
+use bytes::{Buf, BufMut, BytesMut};
+use tokio_util::codec::{Decoder, Encoder};
+
+use crate::proto::wgps::Message;
+
+#[derive(Debug, Default)]
+pub struct WillowCodec;
+
+const MAX_MESSAGE_SIZE: usize = 1024 * 1024 * 1024; // This is likely too large, but lets have some restrictions
+
+impl Decoder for WillowCodec {
+    type Item = Message;
+    type Error = anyhow::Error;
+    fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
+        if src.len() < 4 {
+            return Ok(None);
+        }
+        let bytes: [u8; 4] = src[..4].try_into().unwrap();
+        let frame_len = u32::from_be_bytes(bytes) as usize;
```

Comment:
https://docs.rs/bytes/latest/bytes/buf/trait.Buf.html#method.get_u32 is easier

Reply:
:+1: copied this from `iroh-sync` :smile:

Reply:
oh no, that means I wrote this probably 🤣

Reply:
this is gone after refactors (not using tokio codec anymore)